django_celery_token_bucket/decorators.py

```diff
@@ -1,38 +1,44 @@
 from functools import wraps
-from kombu.connection import Connection
-from kombu.entity import Queue
+from typing import Iterable
 
 from django.conf import settings
+from kombu.connection import Connection
+from kombu.entity import Queue
 
 from django_celery_token_bucket.bucket import TokenBucket
 
 
-def rate_limit(token_bucket_name: str, countdown: int = 60, affect_task_retries: bool = False):
+def rate_limit(token_bucket_name: str | Iterable[str], countdown: int = 60, affect_task_retries: bool = False, weight: int = 1):
     def decorator_func(func):
         @wraps(func)
         def function(self, *args, **kwargs):
-            try:
-                token_bucket: TokenBucket = settings.CELERY_TOKEN_BUCKETS[token_bucket_name]
-                token_queue: Queue = token_bucket.get_queue()
-            except KeyError:
-                raise Exception(f"bucket '{token_bucket_name}' is not registered")
+            if type(token_bucket_name) == str:
+                self.token_bucket_name = [token_bucket_name, ]
 
             connection: Connection
             with self.app.connection_for_read() as connection:
-                # bind queue to a channel of our connection
-                token_queue.maybe_bind(channel=connection.default_channel)
 
-                # try to get a message from the queue
-                message = token_queue.get(no_ack=True)
-                if message:
-                    return func(self, *args, **kwargs)
+                for name in token_bucket_name:
```

Comment:
When failing on the second bucket, the token from the first one would be consumed anyways and not be credited. This is not ideal.

Reply:
Removed multiple buckets per decorator in my most recent commit, using multiple decorators is the easier way.